### PR TITLE
Sm update

### PR DESCRIPTION
**Pull request by cbuendia-atos**
## Proposed Changes

  - Includes compliance with new SM requirements, by adding a function on the SessionManagerClient called `updateSessionVariables()`. This  function calls `/sm/new/add` when the update is done on dsResponse 
  - Includes also previous changes that were not merged on previous PR. Those include
       - Fix issue with CSRF filters and HTTPS
       - Include `msToken` with post redirect on return to rm/response

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests
- [ ] Build locally
- [ ] Code format / lint
- [ ] Docs have been reviewed and added

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

Fixes #

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by rcc-atos — APPROVED** [review]

File: src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java [hunk 1]
```diff
@@ -27,25 +34,65 @@ public DataSet loadDatasetBySAML(String dsId, SAMLCredential credential)
...
+			attributes.add(attributeType);
+		}
+	    AttributeSetStatus atrSetStatus = new AttributeSetStatus();
+	    atrSetStatus.setCode(AttributeSetStatus.CodeEnum.OK);
+	    
+		AttributeSet attrSet = new AttributeSet();
+		attrSet.setId(UUID.randomUUID().toString());
+		attrSet.setType(TypeEnum.RESPONSE);
+		attrSet.setIssuer("edugainASms_001");
+		attrSet.setRecipient("CLms001");
```
Comment (miryamvillegas) [inline src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java]: Be aware! "CLms001"

File: src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java [hunk 2]
```diff
@@ -27,25 +34,65 @@ public DataSet loadDatasetBySAML(String dsId, SAMLCredential credential)
...
+			attributeType.setFriendlyName(att.getFriendlyName());
+			attributes.add(attributeType);
+		}
+	    AttributeSetStatus atrSetStatus = new AttributeSetStatus();
+	    atrSetStatus.setCode(AttributeSetStatus.CodeEnum.OK);
+	    
+		AttributeSet attrSet = new AttributeSet();
+		attrSet.setId(UUID.randomUUID().toString());
+		attrSet.setType(TypeEnum.RESPONSE);
+		attrSet.setIssuer("edugainASms_001");
```
Comment (miryamvillegas) [inline src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java]: Be aware! Use an env variable instead?

**Review by rcc-atos — APPROVED** [review]

File: src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java [hunk 3]
```diff
@@ -27,25 +37,99 @@ public DataSet loadDatasetBySAML(String dsId, SAMLCredential credential)
...
+			attributeType.setFriendlyName(att.getFriendlyName());
+			attributes.add(attributeType);
+		}
+	    AttributeSetStatus atrSetStatus = new AttributeSetStatus();
+	    atrSetStatus.setCode(AttributeSetStatus.CodeEnum.OK);
+	    
+		AttributeSet attrSet = new AttributeSet();
+		attrSet.setId(UUID.randomUUID().toString());
+		attrSet.setType(TypeEnum.RESPONSE);
+		attrSet.setIssuer(System.getenv("RESPONSE_SENDER_ID"));
```
Comment (cbuendia-atos) [inline src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java:88]: This one is the updated

File: src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java (hunk 2 above)
Comment (cbuendia-atos) [reply src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java]: I think this comment is on an older commit! Take a look at the message below where I point to the latest status , it seems I already changed that (I don't remember when I did it thought :P)

Comment (miryamvillegas) [reply src/main/java/eu/seal/idp/service/impl/SAMLDatasetDetailsServiceImpl.java]: OK

**Review by miryamvillegas — APPROVED** [review]

**Pull request merged**